src/components/common/categoryPage/categoryPage.tsx

```diff
@@ -0,0 +1,54 @@
+import { Category, Good, LOAD_STATUSES } from "../../../types";
+import {Link, useParams} from "react-router-dom";
+import { GoodCategory } from "../goodCategory";
+
+import {
+  getCategorySelector,
+  getLoadStatusCategories,
+  getLoadStatusProducts,
+  getProductsSelector,
+} from "../../../store";
+import { fetchCategories } from "../../../store/slices/sliceCategory";
+import { useEffect, useMemo, useState } from "react";
+import { useAppDispatch, useAppSelector } from "../../../store/hooks";
+import { Loader } from "../loader";
+
+import { getProducts } from "../../../store/api";
+import { fetchProducts } from "../../../store/slices/sliceProducts";
+
+export const CategoryPage = () => {
+  const dispatch = useAppDispatch();
+  const { category } = useParams();
+  const loadStatusProducts = useAppSelector(getLoadStatusProducts);
+  const categories = useAppSelector(getCategorySelector);
+  const products = useAppSelector(getProductsSelector);
+
+  useEffect(() => {
+    dispatch(fetchCategories());
+    window.scrollTo(0, 0)
+  }, []);
+
+  const sort = useMemo(() => {
+    if(category) {
+      return categories.find((item) => item.type === category);
+    }
+  }, [category]);
```

Comment:
нужно все зависимости указывать, чтобы баги потом не ловить
```suggestion
  }, [category, categories]);
```